lib/charted.js

```diff
@@ -9,16 +9,19 @@ module.exports = function(app, charted_root) {
   if (charted_root === undefined) { charted_root = '/charted/' }
 
   app.use(charted_root + 'get', function (req, res, next) {
-    var get_url = url.parse(req.url, true).query.url
+    var urlObj = url.parse(req.url, true)
 
-    if (!get_url) {
+    if (!urlObj.query.url) {
       res.statusCode = 400
       res.end('Bad Request: no url provided')
-      next()
       return
     }
 
-    request(decodeURIComponent(get_url), function (err, resp, body) {
+    var get_url = urlObj.query.url.match(/:\/\//) ? urlObj.query.url :
+        ('http://' + req.headers.host + '/' +
+         urlObj.query.url.replace(/^\//, ''))
```

Comment:
In the second case, urlObj.query.url is just a relative path. It doesn't seem possible to get the protocol from urlObj and I couldn't find it elsewhere on the request or app objects when using Matador.

It isn't a problem in Express which is the server the Charted demo uses in this Github repo. But Connect (which Matador extends) doesn't seem to try to figure out how the user is talking to the web server, or those headers are stripped before they get passed to Charted.

Let me know if you think there's another way to figure out, or a header I haven't looked at.